From stable-bounces@linux.kernel.org Thu Feb 21 16:35:12 2008
From: Oliver Neukum <oliver@neukum.org>
Date: Fri, 22 Feb 2008 00:35:05 GMT
Subject: USB: fix pm counter leak in usblp
Message-ID: <200802220035.m1M0Z5Xh015119@hera.kernel.org>

From: Oliver Neukum <oliver@neukum.org>

commit 1902869019918411c148c18cc3a22aade569ac9a upstream

if you fail in open() you must decrement the pm counter again.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/usblp.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/class/usblp.c
+++ b/drivers/usb/class/usblp.c
@@ -428,6 +428,7 @@ static int usblp_open(struct inode *inod
 	usblp->rcomplete = 0;
 
 	if (handle_bidir(usblp) < 0) {
+		usb_autopm_put_interface(intf);
 		usblp->used = 0;
 		file->private_data = NULL;
 		retval = -EIO;
